CMakeLists.txt

```diff
@@ -682,37 +675,36 @@ if(WITH_OPTIM)
         list(APPEND ZLIB_ARCH_HDRS ${ARCHDIR}/arm.h)
         list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/armfeature.c)
         if(WITH_NEON)
-            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/adler32_neon.c ${ARCHDIR}/slide_neon.c)
             add_definitions(-DARM_NEON_ADLER32 -DARM_NEON_SLIDEHASH)
-            add_intrinsics_option("${NEONFLAG}")
+            set(NEON_SRCS ${ARCHDIR}/adler32_neon.c ${ARCHDIR}/slide_neon.c)
+            list(APPEND ZLIB_ARCH_SRCS ${NEON_SRCS})
+            add_intrinsics_option("${NEONFLAG}" ${NEON_SRCS})
             if(MSVC)
                 add_definitions(-D__ARM_NEON__)
             endif()
             add_feature_info(NEON_ALDER32 1 "Support NEON instructions in adler32, using \"${NEONFLAG}\"")
             add_feature_info(NEON_SLIDEHASH 1 "Support NEON instructions in slide_hash, using \"${NEONFLAG}\"")
         endif()
         if(WITH_ACLE)
-            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/crc32_acle.c ${ARCHDIR}/insert_string_acle.c)
             add_definitions(-DARM_ACLE_CRC_HASH)
+            set(ACLE_SRCS ${ARCHDIR}/crc32_acle.c ${ARCHDIR}/insert_string_acle.c)
             # For ARM aarch64, we need to check WITH_NEON first
             if("${ARCH}" MATCHES "arm" OR NOT WITH_NEON)
-                add_intrinsics_option("${ACLEFLAG}")
+                add_intrinsics_option("${ACLEFLAG}" ${ACLE_SRCS})
             endif()
+            list(APPEND ZLIB_ARCH_SRCS ${ACLE_SRCS})
             add_feature_info(ACLE_CRC 1 "Support ACLE optimized CRC hash generation, using \"${ACLEFLAG}\"")
         endif()
     elseif(BASEARCH_PPC_FOUND)
         if(WITH_POWER8 AND HAVE_POWER8)
             add_definitions(-DPOWER8)
             add_definitions(-DPOWER_FEATURES)
             add_definitions(-DPOWER8_VSX_ADLER32)
-            set(ZLIB_POWER8_SRCS
-                ${ARCHDIR}/adler32_power8.c
-                ${ARCHDIR}/slide_hash_power8.c)
-            set_source_files_properties(
-                ${ZLIB_POWER8_SRCS}
-                PROPERTIES COMPILE_FLAGS ${POWER8FLAG})
             list(APPEND ZLIB_ARCH_HDRS ${ARCHDIR}/power.h)
-            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/power.c ${ZLIB_POWER8_SRCS})
+            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/power.c)
+            set(POWER8_SRCS ${ARCHDIR}/adler32_power8.c ${ARCHDIR}/slide_hash_power8.c)
+            list(APPEND ZLIB_ARCH_SRCS ${POWER8_SRCS})
+            add_intrinsics_option("${POWER8FLAG}" ${POWER8_SRCS})
```

Comment:
OK.

Reply:
Yes this is just so I can call `add_intrinsics_option` more easily.